components/config-provider/__tests__/memo.test.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react';
+import { mount } from 'enzyme';
+import ConfigProvider from '..';
+import { ConfigConsumer } from '../context';
+
+// https://github.com/ant-design/ant-design/issues/27617
+describe('ConfigProvider', () => {
+  const RenderCounter = ({ config }) => {
+    const [count, setCount] = React.useState(0);
+    React.useEffect(() => setCount(c => c + 1), [config]);
+    return <div id="counter">{count}</div>;
+  };
+
+  const Sibling = () => (
+    <ConfigConsumer>{config => <RenderCounter config={config} />}</ConfigConsumer>
+  );
+
+  it('should not generate new context config when render', () => {
```

Comment:
ConfigConsumer is internal component which may remove if some refactor. Test should focus on the final behavior. You can ref this to create a test case instead: https://codesandbox.io/s/focused-knuth-o9wh6?file=/src/App.js:476-789

Reply:
To be clear. Just remove `ConfigConsumer` in test case : )